src/controller/adminController.ts

```diff
@@ -1,73 +1,119 @@
+        return;
+      }
+
+      const admins = await this.adminService.listarAdmins(adminLogadoId);
+      res.status(200).json({
+        message: 'Admins listados com sucesso.',
+        data: admins
+      });
     } catch (error) {
-      next(error);
+      res
+        .status(error instanceof ErrorHandler ? error.statusCode : 500)
+        .json({ message: error.message || 'Erro ao listar admins.' });
     }
   }
 
-  async buscarAdminPorId(req: Request, res: Response, next: NextFunction) {
+  async buscarAdminPorId(req: Request, res: Response) {
     try {
+      const adminLogadoId = req.user?.id;
+
+      if (!adminLogadoId) {
+        res.status(401).json({ message: 'Admin não autenticado.' });
+        return;
+      }
+
       const id = parseInt(req.params.id, 10);
-      const admin = await this.adminService.buscarAdminPorId(id);
-      res.json(admin);
+      const admin = await this.adminService.buscarAdminPorId(id, adminLogadoId);
+      res.status(200).json({
+        message: 'Admin encontrado com sucesso.',
+        data: admin
+      });
     } catch (error) {
-      next(error);
+      res
+        .status(error instanceof ErrorHandler ? error.statusCode : 500)
+        .json({ message: error.message || 'Erro ao buscar admin.' });
     }
   }
 
-  async atualizarAdmin(req: Request, res: Response, next: NextFunction) {
+  async atualizarAdmin(req: Request, res: Response) {
     try {
-      const id = parseInt(req.params.id, 10);
+      const adminLogadoId = req.user?.id;
 
-      if (req.body.email) {
-        await this.adminService.verificarEmailDuplicado(req.body.email);
+      if (!adminLogadoId) {
+        res.status(401).json({ message: 'Admin não autenticado.' });
+        return;
       }
 
+      const id = parseInt(req.params.id, 10);
       const adminAtualizado = await this.adminService.atualizarAdmin(
         id,
-        req.body
+        req.body,
+        adminLogadoId
       );
-      res.json(adminAtualizado);
+      res.status(200).json({
+        message: 'Admin atualizado com sucesso.',
+        data: adminAtualizado
+      });
     } catch (error) {
-      next(error);
+      res
+        .status(error instanceof ErrorHandler ? error.statusCode : 500)
+        .json({ message: error.message || 'Erro ao atualizar admin.' });
     }
   }
 
-  async deletarAdmin(req: Request, res: Response, next: NextFunction) {
+  async deletarAdmin(req: Request, res: Response) {
     try {
-      const id = parseInt(req.params.id, 10);
-      const adminLogadoId = req.user.id;
+      const adminLogadoId = req.user?.id;
+
+      if (!adminLogadoId) {
+        res.status(401).json({ message: 'Admin não autenticado.' });
+        return;
+      }
 
+      const id = parseInt(req.params.id, 10);
       await this.adminService.deletarAdmin(id, adminLogadoId);
-      res.status(204).send();
+      res.status(204).json({ message: 'Admin excluído com sucesso.' });
```

Comment:
O status 204 é utilizado para sinalizar que o endpoint foi executado com sucesso e não possui conteúdo dentro do endpoint.
Como nesse caso há uma mensagem de exibição, o retorno de status seria de 200, pois ele tem no body um message: 'Admin excluído com sucesso.'.


[Forum explicando status 200, 204 e 400](https://medium.com/localizalabs/404-204-200-qual-status-a-api-deve-retornar-quando-a-resposta-for-vazia-e4b153936398)